tlsconfig/config_test.go

```diff
@@ -377,6 +377,70 @@ func TestConfigServerExclusiveRootPools(t *testing.T) {
 	}
 }
 
+// If a valid minimum version is specified in the options, the server's
+// minimum version should be set accordingly
+func TestConfigServerTLSMinVersionIsSetBasedOnOptions(t *testing.T) {
+	versions := []uint16{
+		tls.VersionTLS11,
+		tls.VersionTLS12,
+	}
+	tempDir := makeTempDir(t)
+	defer os.RemoveAll(tempDir)
+	key, cert := generateCertAndKey(t, tempDir)
```

Comment:
Just a note for myself or whoever wants to fix this, but we should be using generated certs/keys for tests and not regenerate every time. Should be done in another PR.